programs/bench.c

```diff
@@ -71,6 +72,7 @@ static U32 g_compressibilityDefault = 50;
 ***************************************/
 #define DISPLAY(...)         fprintf(stderr, __VA_ARGS__)
 #define DISPLAYLEVEL(l, ...) if (g_displayLevel>=l) { DISPLAY(__VA_ARGS__); }
+#define DISPLAYLEVELP(l, ...) if (g_displayLevel>=l && printable) { DISPLAY(__VA_ARGS__); }
```

Comment:
That feels strange.
This macro now depends on both `g_displayLevel` and some local variable `printable` to exist within scope.

I've no doubt you can make it to work in this patch, but this is a recipe for problems in future patches.

Consider having `displayLevel` a direct argument of the function.
This way, setting to `0` is enough to disable any display.

This may enter in competition with currently defined `g_displayLevel`.
One cheap way to get it to work is to start your function with `g_displayLevel = displayLevel`.
Everything else stay the same.

A more involved but proper way is to remove `g_displayLevel` altogether.
Now that `bench.c` is no longer a stand-alone module, it becomes more desirable.
You would then have to take care of proper propagation of `displayLevel` argument.